tests/ui/while_let_loop.rs

```diff
@@ -22,6 +22,11 @@ fn main() {
         break;
     }
 
+    loop {
+        //~^ while_let_loop
+        let Some(_x) = y else { break };
+    }
+
```

Comment:
Can we also include some test cases where the else block has side effects?

i.e.

```rust

    loop {
        //~^ while_let_loop
        let Some(_x) = y else {
            side_effect();
            break 
        };
    }
```